Error in 'Tokens' tab when loading large conversations.json: special token <|endoftext|> not allowed

Hi, 

I loaded conversations.json (106MB) and everything works smoothly except for the 'Tokens' tab. When I go there, it is processing for a few seconds, and later on, it shows the error in the Chrome console: 

`app-index.tsx:25 Error calculating token usage: Error: The text contains a special token that is not allowed: <|endoftext|>
    at __wbindgen_error_new (tiktoken_bg.js:417:1)
    at 39d59c54d2ebb795.wasm:0x2a056
    at 39d59c54d2ebb795.wasm:0x7d32f
    at Tiktoken.encode (tiktoken_bg.js:288:1)
    at OptimizedTokenCounter.countTokens (ChatGPTDataAnalysis.ts:112:20)
    at eval (ChatGPTDataAnalysis.ts:670:46)
    at Array.forEach (<anonymous>)
    at eval (ChatGPTDataAnalysis.ts:646:41)
    at Array.forEach (<anonymous>)
    at ChatGPTDataAnalysis.getMonthlyModelWiseTokenUsage (ChatGPTDataAnalysis.ts:645:13)
    at eval (page.tsx:215:46)`

Maybe a bug? 

I had cloned the repo yesterday:  SHA-1: e80a27ffbaf11bec7c585b877322220f5d50b371  (17. April 2025 6:16:53)

--- a/lib/ChatGPTDataAnalysis.ts
+++ b/lib/ChatGPTDataAnalysis.ts
@@ -94,7 +94,7 @@ class OptimizedTokenCounter {
   private getEncoder(modelSlug: string): any {
     if (!this.encoders[modelSlug]) {
       const tiktokenModel = this.getMatchingTiktokenModel(modelSlug);
-      this.encoders[modelSlug] = encoding_for_model(tiktokenModel);
+      this.encoders[modelSlug] = encoding_for_model(tiktokenModel,{ '<|endoftext|>': 50256 });
     }
     return this.encoders[modelSlug];
   }
@@ -109,7 +109,7 @@ class OptimizedTokenCounter {
 
 public countTokens(text: string, modelSlug: string): number {
     const encoder = this.getEncoder(modelSlug);
-    return encoder.encode(text, { allowed_special: { '<|endoftext|>': true } }).length;
+    return encoder.encode(text,{ allowed_special: new Set(['<|endoftext|>']) }).length;
 }
 
 

--- a/lib/ChatGPTDataAnalysis.ts
+++ b/lib/ChatGPTDataAnalysis.ts
@@ -109,7 +109,7 @@ class OptimizedTokenCounter {
 
   public countTokens(text: string, modelSlug: string): number {
     const encoder = this.getEncoder(modelSlug);
-    return encoder.encode(text).length;
+    return encoder.encode(text,allowed_special={'<|endoftext|>'}).length;
   }
 
   public freeEncoders(): void {

--- a/lib/ChatGPTDataAnalysis.ts
+++ b/lib/ChatGPTDataAnalysis.ts
@@ -107,10 +107,11 @@ class OptimizedTokenCounter {
     return "gpt-3.5-turbo";
   }
 
-  public countTokens(text: string, modelSlug: string): number {
+public countTokens(text: string, modelSlug: string): number {
     const encoder = this.getEncoder(modelSlug);
-    return encoder.encode(text,allowed_special={'<|endoftext|>'}).length;
-  }
+    return encoder.encode(text, { allowed_special: { '<|endoftext|>': true } }).length;
+}
+
 
   public freeEncoders(): void {
     Object.values(this.encoders).forEach(encoder => encoder.free());

--- a/lib/ChatGPTDataAnalysis.ts
+++ b/lib/ChatGPTDataAnalysis.ts
@@ -109,7 +109,7 @@ class OptimizedTokenCounter {
 
 public countTokens(text: string, modelSlug: string): number {
     const encoder = this.getEncoder(modelSlug);
-    return encoder.encode(text,{ allowed_special: new Set(['<|endoftext|>']) }).length;
+    return encoder.encode(text,"all").length;
 }
 
 

